packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx

```diff
@@ -59,14 +78,55 @@ export const fiatExchangesOptionsScreenOptions = ({
 }
 
 function FiatExchangeOptions({ route, navigation }: Props) {
+  const [providerLocalAmount, setProviderLocalAmount] = React.useState<BigNumber>()
+  const [providerlocalCurrency, setProviderLocalCurrency] = React.useState<LocalCurrencyCode>()
+  // All currency exchange calculations handled here as
+  // different providers may provide different fiat, CELO and cUSD rates
+
+  const localCurrency = useSelector(getLocalCurrencyCode)
   const { amount } = route.params
+
   const goToProvider = (screen: keyof StackParamList) => {
-    return () => navigation.navigate(screen, { amount })
+    return () =>
+      navigation.navigate(screen, {
+        localAmount: features.CUSD_MOONPAY_ENABLED
+          ? amount
+          : providerLocalAmount || new BigNumber(0),
+        currencyCode: providerlocalCurrency || FALLBACK_CURRENCY,
+      })
   }
   const onDismiss = () => {
     navigation.setParams({ isExplanationOpen: false })
   }
 
+  const dispatch = useDispatch()
+  React.useEffect(() => {
```

Comment:
Nit, if you import `useEffect` directly then you can use it without going through React, similar to how you're using useDispatch in live above.

Reply:
👍 updated for useEffect and useState